src/megaapi_impl.cpp

```diff
@@ -1345,6 +1345,24 @@ bool MegaApiImpl::isIndexing()
     return indexing;
 }
 
+bool MegaApiImpl::isSyncing()
+{
+    if (!client->syncs.size())
+    {
+        return false;
+    }
+    SdkMutexGuard g(sdkMutex);
+
+    for (auto & sync : client->syncs)
+    {
+        if (sync->localroot->ts == TREESTATE_SYNCING)
```

Comment:
Not sure, but you might want to consider the abnormal state `TREESTATE_PENDING` here too, since it's also syncing, but waiting for stable `mtime`/`size` or waiting for the upload to complete and put the node.

It all depends on the planned usage for this new function.

Reply:
agree